### PR TITLE
Add print interface function

### LINKED ISSUE #70: PrintInterface()

### PR DESCRIPTION
**Pull request by lukasz-kukulka**

### CODE REVIEW

**Review by BarTes8 — APPROVED** [review]

File: shm/inc/Time.hpp [hunk 1]
```diff
@@ -12,7 +12,7 @@ class Time {
     size_t getElapsedTime() const;
 
 private:
-    size_t elapsedTime_{};
+    size_t elapsedTime_{ 0 };
```
Comment (mariuszlisowski) [inline shm/inc/Time.hpp]: redundant zero

File: shm/src/Game.cpp [hunk 2]
```diff
@@ -5,20 +5,21 @@
...
+Game::Game(size_t money, size_t gameDays, size_t finalGoal)
     : money_(money)
-    , game_days_(game_days)
-    , final_goal_(final_goal)
+    , gameDays_(gameDays)
+    , finalGoal_(finalGoal)
     , time_(std::make_unique<Time>())
 {
-    current_day_ = time_->getElapsedTime();
+    gameDays_ -= time_->getElapsedTime();
```
Comment (mariuszlisowski) [inline shm/src/Game.cpp]: no, gameDays_ is already initialised above. currentDay_ should start with 0 (time_->getElapsedTime should return 0 when Game is initialised)

File: shm/inc/Game.hpp [hunk 3]
```diff
@@ -9,18 +9,19 @@
...
 private:
     void printWelcomeScreen();
     void printMenu();
+    void printIntenface();
     
     size_t money_ {};
-    size_t game_days_ {};
-    const size_t final_goal_ {};
-    size_t current_day_;
+    size_t gameDays_ {};
```
Comment (mariuszlisowski) [inline shm/inc/Game.hpp]: gameDays_ stores the total length of the game and is set when the game starts

File: shm/src/Game.cpp [hunk 4]
```diff
@@ -5,20 +5,21 @@
...
     ship_ = std::make_unique<Ship>(1, 25, 100, nullptr);
     player_ = std::make_unique<Player>(std::move(ship_), 100, 10000);
 }
 
 void Game::startGame() {
     printWelcomeScreen();
-    while (game_days_ > current_day_) {
+    printIntenface();
+    while (gameDays_ > 0) {
```
Comment (mariuszlisowski) [inline shm/src/Game.cpp]: infinite loop - time goes forward (operator++) as gameDays_ indicated currentDay (??)

**Review by mariuszlisowski — CHANGES_REQUESTED** [review]

File: shm/inc/Game.hpp [hunk 5]
```diff
@@ -9,18 +9,19 @@
...
     void startGame();
 
 private:
     void printWelcomeScreen();
     void printMenu();
+    void printIntenface();
     
     size_t money_ {};
-    size_t game_days_ {};
-    const size_t final_goal_ {};
```
Comment (mariuszlisowski) [inline shm/inc/Game.hpp]: no need to initialize here as initialized in constructor (remove {})

File: shm/src/Game.cpp [hunk 6]
```diff
@@ -62,3 +63,13 @@ void Game::printMenu() {
     std::cout << "#" << std::setfill(' ')  << std::setw (30) << "#\n";
     std::cout << "#" << std::setfill('-')  << std::setw (31) << "#\n\n";
 }
+
+void Game::printIntenface() {
+    std::cout << std::setw (99) << std::setfill('#') << "\n";
+    std::cout << "#" << std::setfill(' ') << std::setw (97) << "#" << "\n";
+    std::cout << "#" << std::setfill(' ') << std::setw (15) << "YOUR MONEY: " << std::setw (8) << std::setfill('0') << money_;
+    std::cout << std::setfill(' ') << std::setw (30) << "YOUR GOAL: " << std::setw (8) << std::setfill('0') << finalGoal_;
+    std::cout << std::setfill(' ') << std::setw (30) << "DAYS LEFT: " << std::setw (3) << std::setfill('0') << currentDay_ << std::setfill(' ') << std::setw (4) << "#" "\n";
```
Comment (mariuszlisowski) [inline shm/src/Game.cpp]: days left = gameDays - currentDay

File: shm/src/Game.cpp [hunk 7]
```diff
@@ -5,20 +5,21 @@
...
+Game::Game(size_t money, size_t gameDays, size_t finalGoal)
     : money_(money)
-    , game_days_(game_days)
-    , final_goal_(final_goal)
+    , gameDays_(gameDays)
+    , finalGoal_(finalGoal)
     , time_(std::make_unique<Time>())
 {
-    current_day_ = time_->getElapsedTime();
+    size_t currentDay_ = time_->getElapsedTime();
```
Comment (mariuszlisowski) [inline shm/src/Game.cpp]: no, variable shadows private data member - let's delete the line as it is already initialised in Game.hpp (with zero which is the first day)

**Review by mariuszlisowski — CHANGES_REQUESTED** [review]

**Review by mariuszlisowski — APPROVED** [review]

**Pull request merged**